### PR TITLE
Wayland rewrite

### PR DESCRIPTION
**Pull request by atomnuker**
The wayland code was written more than 4 years ago when wayland wasn't even at version 1.0. This commit rewrites everything in a more modern way, switches to using the new xdg v6 shell interface which solves a lot of bugs and makes mpv tiling-friedly, adds support for drag and drop, adds support for touchscreens and adds support for KDE's server decorations protocol.
Currently supported touchscreen events: two finger move, one finger scale (touch the edge of the window) and one-finger tapping to click.

It does not yet use the frame callback as a main rendering loop, this will happen with a later commit.

The protocol files are included in the repo because waf can't seem to use them from /usr/share/wayland-protocols.

I agree that my changes can be relicensed to LGPL 2.1 or later.

### CODE REVIEW

File: video/out/wayland_common.c [hunk 1]
```diff
@@ -18,49 +15,222 @@
...
+
+    if (!mp_input_test_dragging(wl->vo->input_ctx, wl->mouse_x, wl->mouse_y) &&
+        (button == MP_MBTN_LEFT) && (state == MP_KEY_STATE_DOWN))
+        window_move(wl, serial);
+}
+
+static void pointer_handle_axis(void *data, struct wl_pointer *wl_pointer,
+                                uint32_t time, uint32_t axis, wl_fixed_t value)
+{
+}
```
Comment (SirCmpwn) [inline video/out/wayland_common.c]: Don't you need to implement this?

Comment (atomnuker) [reply video/out/wayland_common.c]: Yep, I forgot to.
Implemented.

**Comment by atomnuker** [comment]
Thanks for the review, pushed

**Comment by vanvugt** [comment]
Tangentially, if anyone is still using mpv 0.27 and earlier on Gnome Shell then this shell fix solves clicking/moving/tiling issues too (without needing the rewrite in mpv 0.28):
  https://bugzilla.gnome.org/show_bug.cgi?id=790309
With some luck that will make it into mutter 3.26.3.

**Pull request closed without merging**